src/libstore/local-store.cc

```diff
@@ -212,16 +212,15 @@ LocalStore::LocalStore(
 
     /* Ensure that the store and its parents are not symlinks. */
     if (!settings.allowSymlinkedStore) {
-        Path path = realStoreDir;
-        struct stat st;
-        while (path != "/") {
-            st = lstat(path);
-            if (S_ISLNK(st.st_mode))
+        std::filesystem::path path = realStoreDir.get();
+        std::filesystem::path root = path.root_path();
+        while (path != root) {
+            if (std::filesystem::is_symlink(path))
```

Comment:
Original comments on this were:

> Would be nice to wrap is_symlink in an try-catch and re-throw a nix error instead the original exception that std::filesystem throws because those actually will have an error trace added when the exception is bubbled up.

And:

> The function rethrowExceptionAsError() can be used for that. (Should be moved from local-derivation-goal.cc to libutil, probably.)

Reply:
From reading the [filesystem API docs](https://en.cppreference.com/w/cpp/filesystem/status#Exceptions) and the source for [GCC libstdc++](https://github.com/gcc-mirror/gcc/blob/bcbbce233bb3989673e099aace7a876d569d8d80/libstdc%2B%2B-v3/src/filesystem/ops.cc#L1270-L1274), I can see that bad memory allocation should be the only exception. I don't think it makes sense to try/catch here.